src/test/java/com/commercetools/sync/commons/FakeClient.java

```diff
@@ -0,0 +1,50 @@
+package com.commercetools.sync.commons;
+
+import io.sphere.sdk.client.SphereApiConfig;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.client.SphereRequest;
+import io.sphere.sdk.utils.CompletableFutureUtils;
+
+import javax.annotation.Nonnull;
+import java.util.concurrent.CompletionStage;
+
+public class FakeClient<T> implements SphereClient {
+
+    private boolean isExecuted = false;
+    private T mockResult;
+    private Throwable mockException;
+
+    public FakeClient(@Nonnull final T mockResult) {
+        this.mockResult = mockResult;
+    }
+
+    public FakeClient(@Nonnull final Throwable mockException) {
+        this.mockException = mockException;
+    }
+
+    @Override
+    public <T> CompletionStage<T> execute(final SphereRequest<T> sphereRequest) {
+        isExecuted = true;
+        if (mockResult != null) {
+            return CompletableFutureUtils.successful((T)mockResult);
+        } else if (mockException != null) {
+            return CompletableFutureUtils.failed(mockException);
+        } else {
+            return CompletableFutureUtils.successful(null);
```

Comment:
This part of if branch is never reachable, you can remove it.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/622/commits/8b1a907d7a65a2707752e4073fdac629715c7319

Reply:
```
        if (mockResult != null) {
            return CompletableFutureUtils.successful((T)mockResult);
        } else {
            return CompletableFutureUtils.failed(mockException);
        }
```

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/622/commits/9d153ac27794cffc6c1ed8a685c69fe300919f08